akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -78,6 +77,22 @@ object Sink {
     new Sink(scaladsl.Sink.asPublisher(fanout == AsPublisher.WITH_FANOUT))
 
   /**
+   * A `Sink` that materializes into a [[org.reactivestreams.Publisher]] with additional cancel method.
+   * The difference from [[Sink.asPublisher]] is that sink is still doing backpressure without publishers
+   * connected and does not finalize stream when last publisher cancels. It completes state when cancel
+   * is called or complete came from upsream.
+   *
+   * The materialized `Publisher` will support multiple `Subscriber`s and
+   * the size of the `inputBuffer` configured for this stage becomes the maximum number of elements that
+   * the fastest [[org.reactivestreams.Subscriber]] can be ahead of the slowest one before slowing
+   * the processing down due to back pressure.
+   *
+   * @see [[#asPublisher]]
+   */
+  def asPublisher[T](fanout: Boolean, finalizeOnLastSubscriptionCompletion: Boolean): Sink[T, Publisher[T]] =
```

Comment:
I think I'd rather see a different factory name that always creates it as non-finalizing-on-last-subscription-completion than having a boolean parameter where one value makes it behave exactly like the existing `asPublisher` method.

I don't have a good name for it from the top of my head though, `asDurablePublisher`, `asLongLivedPublisher`, `asStablePublisher` ?